quickwit-cluster/src/cluster.rs

```diff
@@ -88,8 +91,24 @@ pub struct Member {
     pub is_self: bool,
 }
 
+/// Cluster meant to manage Quickwit's cluster formation and its events.
+#[async_trait]
+pub trait Cluster: Send + Sync + 'static {
```

Comment:
What is the point of this cluster trait (as opposed to just having a `EpidemicCluster` implementing these public methods?

Reply:
At the moment it only supports cluster formation using Artillery, but I thought it would be useful in the future to have a trait that would make it easy to use different cluster formations.

Reply:
This may be controversial, but I usually don't add abstractions if not necessary, since working with abstractions lowers readability imo.

Reply:
@fulmicoton @PSeitz Thanks. I'll revert it.